dhis-2/dhis-api/src/main/java/org/hisp/dhis/webapi/controller/event/webrequest/PagingAndSortingCriteriaAdapter.java

```diff
@@ -88,4 +110,44 @@ private boolean isSkipPaging()
         return Optional.ofNullable( skipPaging )
             .orElse( false );
     }
+
+    @Override
+    public List<OrderCriteria> getOrder()
+    {
+        if ( getAllowedOrderingFields().isEmpty() )
+        {
+            return dtoNameToDatabaseNameTranslator.apply( order );
+        }
+
+        Map<Boolean, List<OrderCriteria>> orderCriteriaPartitionedByAllowance = CollectionUtils.emptyIfNull( order )
+            .stream()
+            .collect(
+                partitioningBy( this::isAllowed ) );
+
+        CollectionUtils.emptyIfNull( orderCriteriaPartitionedByAllowance.get( false ) )
+            .forEach( disallowedOrderFieldConsumer() );
+
+        return dtoNameToDatabaseNameTranslator.apply( orderCriteriaPartitionedByAllowance.get( true ) );
+    }
+
+    private boolean isAllowed( OrderCriteria orderCriteria )
+    {
+        return getAllowedOrderingFields().contains( orderCriteria.getField() );
+    }
+
+    protected Consumer<? super OrderCriteria> disallowedOrderFieldConsumer()
+    {
+        return orderCriteria -> log.warn( "Ordering by " + orderCriteria.getField() + " is not supported" );
+    };
+
```

Comment:
we can remove this ; it is also reviewed by SonarQube

Reply:
fixed